src/pylibsshext/channel.pyx

```diff
@@ -46,6 +46,7 @@ cdef int _process_outputs(libssh.ssh_session session,
 
 cdef class Channel:
     def __cinit__(self, session):
+        self._session = session
```

Comment:
@pbrezina is this the place with the actual fix for the SEGFAULT?

Reply:
Is this going to fix https://github.com/ansible/pylibssh/issues/57? If yes, then plz symlink `57.bugfix.rst` to `576.bugfix.rst`.

Reply:
It's likely, yes. But I can't tell without a coredump.

This is a fix for segfault I seen, test for it is in the test_destructor. The problem is that you open channel in cinit, then close it in destructor. To close it, you need to have valid session object. But if you rely on the destructor to close the channel, then it may happen that Session is gc'd first, therefore libssh_session is freed before you try to close the channel.

This patch makes sure that we keep reference to the python session object (not only to libssh session object) and therefore Channel is always gc'd before Session.

Reply:
Alright, thank you for the explanation! I'm marking this thread as resolved.